File: BAAPCoders/HerSafety/app/src/main/java/com/s/hersafety/LoginActivity.java
package com.s.hersafety;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.airbnb.lottie.LottieAnimationView;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.s.hersafety.Model.Location;
import com.s.hersafety.Model.PoliceStation;
import com.s.hersafety.Model.PoliceStationLocation;

public class LoginActivity extends AppCompatActivity {
    Button btnLogin;
    LottieAnimationView animationView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);initialiseVariables();

        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                animationView.setAnimation("phonesuccess.json");

                for (int i = 0; i<5; i++) {
                    PoliceStation policeStation = new PoliceStation();
                    policeStation.setIncharge("TestIncharge"+i);
                    policeStation.setDistance("0m");
                    policeStation.setArea("AdarshNagar"+i);
                    policeStation.setLocation(new PoliceStationLocation("27.0299139",
                            "75.8936901",
                            "Arya College, Jaipur"));
                    DatabaseReference reference = FirebaseDatabase.getInstance().getReference("police");

                    reference.child("location" + i).setValue(policeStation);
                }
                startActivity(new Intent(LoginActivity.this,MainActivity.class));
            }
        });
    }

    private void initialiseVariables() {
        btnLogin = findViewById(R.id.btn_otp);
        animationView = findViewById(R.id.animation_login);
    }


    //final Handler handler = new Handler();
    //                handler.postDelayed(new Runnable() {
    //                    @Override
    //                    public void run() {
    //                        startActivity(new Intent(LoginActivity.this,MainActivity.class));
    //                    }
    //                }, 2000);
}

File: HPS/track/app/src/main/java/com/logicaldp/track/LoginActivity.kt
package com.logicaldp.track

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import kotlinx.android.synthetic.main.signin.*


class LoginActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        val mAuth = FirebaseAuth.getInstance()
        ButtonSignin.setOnClickListener {
            if (EditTextEmail.text.toString() != "")
                if (EditTextPassword.text.toString() != "") {
                    mAuth.signInWithEmailAndPassword(EditTextEmail.text.toString(), EditTextPassword.text.toString()).addOnCompleteListener{
                        if(it.isSuccessful) {
                            val user = mAuth.currentUser
                            val database = FirebaseDatabase.getInstance().getReference("users")
                            database.child(mAuth.uid.toString()).child("type").addListenerForSingleValueEvent(object : ValueEventListener {
                                override fun onDataChange(dataSnapshot: DataSnapshot) {
                                    val value =
                                        dataSnapshot.getValue(String::class.java)!!
                                    when(value){
                                        "gaurdian" ->{
                                            val intent = Intent(this@LoginActivity, MapActivity::class.java)
                                            intent.putExtra("gaurdian", true)
                                            startActivity(intent)
                                        }
                                        "driver" ->{
                                            val intent = Intent(this@LoginActivity, MapActivity::class.java)
                                            intent.putExtra("driver", true)
                                            startActivity(intent)
                                        }
                                        else ->{
                                            val intent = Intent(this@LoginActivity, MainActivity::class.java)
                                            intent.putExtra("admin", true)
                                            startActivity(intent)
                                        }
                                    }
                                }
                                override fun onCancelled(error: DatabaseError) {
                                }
                            })
//                                database.child(user!!.uid).child("type")
//
//                            startActivity(Intent(this, MainActivity::class.java))

                            val intent = Intent(this@LoginActivity, MapActivity::class.java)
                            intent.putExtra("driver", true)
                            startActivity(intent)
                        }else{
                            Toast.makeText(this, "Authentication Failed",Toast.LENGTH_LONG).show()
                        }
                    }
                } else {
                    Toast.makeText(this, "Please enter Password",Toast.LENGTH_LONG).show()
                }
            else{
                Toast.makeText(this, "Please enter Email",Toast.LENGTH_LONG).show()
            }
        }
    }
}
